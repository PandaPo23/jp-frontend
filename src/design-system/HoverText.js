import PropTypes from 'prop-types';

import dynamicVariant from '../utils/DynamicVariant';
import system from '../utils/System';
import Text from './Text';

const hoverVariant = dynamicVariant({
  key: 'hovers',
  prop: 'hover',
});

const HoverText = system(
  'HoverText',
  {
    extend: Text,
    bg: 'surface',
    color: 'on.surface',
    hover: 'links.darkenBg',
  },
  hoverVariant
);

HoverText.propTypes = {
  hover: PropTypes.any,
};

export default HoverText;
