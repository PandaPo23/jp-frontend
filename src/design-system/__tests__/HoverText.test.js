import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import HoverText from '../HoverText';

describe('HoverText component', () => {
  it('should render as HoverText', () => {
    const wrapper = mount(<HoverText>Text</HoverText>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
