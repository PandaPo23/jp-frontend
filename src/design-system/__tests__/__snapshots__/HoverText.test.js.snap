// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoverText component should render as HoverText 1`] = `
.emotion-0 {
  color: on.surface;
  background-color: surface;
}

<HoverText
  bg="surface"
  color="on.surface"
  hover="links.darkenBg"
>
  <div
    className="emotion-0"
  >
    Text
  </div>
</HoverText>
`;
